### PR TITLE
Fix capability

### PR DESCRIPTION
**Pull request by rjeczalik**
Fix for broken Capability marshalling/unmarshalling on master.

### CODE REVIEW

File: rest/auth.go [hunk 1]
```diff
@@ -35,29 +50,31 @@ type tokenResponse struct {
...
-	params := []string{
-		t.ID,
-		strconv.Itoa(t.TTL),
-		t.Capability,
-		t.ClientID,
-		strconv.Itoa(int(t.Timestamp)),
-		t.Nonce,
-	}
-	s := strings.Join(params, "\n") + "\n"
+var nl = []byte{'\n'}
```
Comment (kouno) [inline rest/auth.go]: Shouldn't this be within the `Sign` function anyway?

File: rest/auth.go [hunk 2]
```diff
@@ -35,29 +50,31 @@ type tokenResponse struct {
...
+	mac.Write([]byte(strconv.Itoa(t.TTL)))
+	mac.Write(nl)
+	mac.Write([]byte(t.Capability.String()))
+	mac.Write(nl)
+	mac.Write([]byte(t.ClientID))
+	mac.Write(nl)
+	mac.Write([]byte(strconv.Itoa(int(t.Timestamp))))
+	mac.Write(nl)
+	mac.Write([]byte(t.Nonce))
+	mac.Write(nl)
```
Comment (kouno) [inline rest/auth.go]: Hum... isn't replacing the array and the string a micro optimization? Also it's less readable IMO.

**Comment by kouno** [comment]
Looks good :+1:.

File: rest/auth.go (hunk 2 above)
Comment (rjeczalik) [inline rest/auth.go]: I was rather aiming for making this readable that optimising - for me series of Writes are easier to follow than []string + strings.Join(..., "\n") + single write.

If it's less readable for you, I'll revert this change.

File: rest/auth.go (hunk 1 above)
Comment (rjeczalik) [inline rest/auth.go]: The nl will be gone after revert.

File: rest/auth.go (hunk 2 above)
Comment (kouno) [inline rest/auth.go]: It would be as readable if that wasn't interleaved with `nl` :).

**Comment by rjeczalik** [comment]
@kouno Please take a look.

File: rest/auth.go [hunk 3]
```diff
@@ -35,29 +51,23 @@ type tokenResponse struct {
...
-
 	mac := hmac.New(sha256.New, []byte(secret))
-	mac.Write([]byte(s))
-
+	fmt.Fprintln(mac, t.ID)
+	fmt.Fprintln(mac, t.TTL)
+	fmt.Fprintln(mac, t.Capability.String())
+	fmt.Fprintln(mac, t.ClientID)
+	fmt.Fprintln(mac, t.Timestamp)
+	fmt.Fprintln(mac, t.Nonce)
```
Comment (kouno) [inline rest/auth.go]: Nice :+1:.

**Comment by kouno** [comment]
Nice work.

File: rest/auth.go (hunk 2 above)
Comment (mattheworiordan) [inline rest/auth.go]: It's all ugly to me ;)

File: rest/auth.go (hunk 3 above)
Comment (mattheworiordan) [inline rest/auth.go]: Ahhh, that's nice ;) :smile:

**Pull request merged**